tests/unit/amo/components/TestInstallButtonWrapper.js

```diff
@@ -0,0 +1,239 @@
+import * as React from 'react';
+
+import GetFirefoxButton, {
+  GET_FIREFOX_BUTTON_TYPE_ADDON,
+} from 'amo/components/GetFirefoxButton';
+import InstallButtonWrapper, {
+  InstallButtonWrapperBase,
+} from 'amo/components/InstallButtonWrapper';
+import { setInstallState } from 'core/actions/installations';
+import AMInstallButton from 'core/components/AMInstallButton';
+import { CLIENT_APP_FIREFOX, INSTALLED, UNKNOWN } from 'core/constants';
+import { createInternalAddon } from 'core/reducers/addons';
+import { createInternalVersion, loadVersions } from 'core/reducers/versions';
+import {
+  createContextWithFakeRouter,
+  createFakeLocation,
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeInstalledAddon,
+  shallowUntilTarget,
+  userAgentsByPlatform,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const render = (props = {}) => {
+    return shallowUntilTarget(
+      <InstallButtonWrapper
+        addon={createInternalAddon(fakeAddon)}
+        defaultInstallSource="some-install-source"
+        location={createFakeLocation()}
+        store={store}
+        {...props}
+      />,
+      InstallButtonWrapperBase,
+      {
+        shallowOptions: createContextWithFakeRouter(),
+      },
+    );
+  };
+
+  const _loadVersions = ({ addon = fakeAddon } = {}) => {
+    store.dispatch(
+      loadVersions({
+        slug: addon.slug,
+        versions: [
+          {
+            ...addon.current_version,
+          },
+        ],
+      }),
+    );
+  };
+
+  const _dispatchClientMetadata = (params = {}) => {
+    return dispatchClientMetadata({
+      store,
+      userAgent: userAgentsByPlatform.mac.firefox57,
+      ...params,
+    });
+  };
+
+  it('for Firefox, calls getClientCompatibility to determine the compatibility', () => {
+    const addon = fakeAddon;
+
+    _loadVersions({ addon });
+
+    const clientApp = CLIENT_APP_FIREFOX;
+    const _getClientCompatibility = sinon.mock().returns({
+      compatible: true,
+    });
+
+    _dispatchClientMetadata({
+      clientApp,
+    });
+
+    render({
+      _getClientCompatibility,
+      addon: createInternalAddon(addon),
+      store,
+    });
+
+    sinon.assert.calledWith(_getClientCompatibility, {
+      addon: createInternalAddon(addon),
+      clientApp,
+      currentVersion: createInternalVersion(addon.current_version),
+      userAgentInfo: store.getState().api.userAgentInfo,
+    });
+  });
+
+  it('when not Firefox, does not call getClientCompatibility', () => {
```

Comment:
maybe ~ "it("does not call getClientCompatibility when the it's not a firefox client", () =>`